src/VirtualShield.cpp

```diff
@@ -782,13 +791,19 @@ int VirtualShield::directToSerial(const char* cmd)
 /// <returns>Zero if no error, negative if an error.</returns>
 int VirtualShield::sendFlashStringOnSerial(const char* flashStringAdr, int start, bool encode) const
 {
-	unsigned char dataChar = 0;
-	const int actualStart = start < 0 ? 0 : start;
+	char dataChar = '\0';
+	const size_t actualStart = start < 0 ? 0 : start;
 	const bool isFormatted = start > DEFAULT_LENGTH;
 
-	for (int i = actualStart; i < strlen_PF((uint_farptr_t)flashStringAdr); i++)
+#if defined(_WINDOWS_)
```

Comment:
Add the far pointer PF support to pgmspace.h in lightning. https://github.com/ms-iot/lightning/blob/develop/source/avr/pgmspace.h
then remove this ifdef. It should just work in lightning.